### PR TITLE
fix: dropdown should focus on trigger when context menu closes;

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
Improved Dropdown accessibility.

- Dropdown should focus on trigger when context menu closes (sample reference: https://pauljadam.com/demos/bootstrap-dropdown.html );
- Updated random id generation with react 18 `useId()`;

## Manual Testing

- [ ] Checked on Chrome;
- [ ] Checked on Firefox;

### Preview domain
https://fix-dropdown-accessibility.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/fields/Dropdown.tsx [hunk 1]
```diff
@@ -123,6 +126,12 @@ export function Dropdown({
 
   const fullScreen = openFullScreen ?? isMobile;
 
+  useLayoutEffect(() => {
+    if (wasVisible === 'true' && !isVisible) {
+      triggerRef?.current?.focus?.();
+    }
+  }, [isVisible]);
```
Comment (sshanzel) [inline packages/shared/src/components/fields/Dropdown.tsx:133]: I think we can avoid the additional hooks but just adding the `triggerRef?.current?.focus?.();` at the `onHidden` callback of the `Menu` component.

Comment (ilasw) [reply packages/shared/src/components/fields/Dropdown.tsx:133]: Hi @sshanzel, thanks for comment!  I've tried, but unfortunately `onHidden` is called on the very first render of the `Menu` component, resulting in an autofocus behavior on the last dropdown on the page :(

**Review by rebelchris — APPROVED** [review]
Thanks for the pro-active fix 🙏
If we can't avoid the layoutEffect (which probably we can't due to focus state) I'm happy with it.

**Review by nensidosari — APPROVED** [review]
🚀

File: packages/shared/src/components/fields/Dropdown.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/Dropdown.tsx:133]: Oh, then I think there is something wrong with the callback. Anyways, we will replace our context menu completely in the future. I think spending more time to check why that happens is probably unnecessary. Should be good to go here! 🥳

**Pull request merged**